plugins/fs/fs.c

```diff
@@ -108,10 +109,14 @@ run(LuastatusPluginData *pd, LuastatusPluginRunFuncs funcs)
         lua_State *L = funcs.call_begin(pd->userdata);
         lua_createtable(L, 0, p->paths.size);
         for (size_t i = 0; i < p->paths.size; ++i) {
-            const char *path = p->paths.data[i];
-            if (push_for(pd, L, path)) {
-                lua_setfield(L, -2, path);
+            glob_t gbuf = {0};
+            glob(p->paths.data[i], 0, NULL, &gbuf);
```

Comment:
Why don’t you check the return value of `glob()`?

You would probably want to pass `GLOB_NOCHECK` and `GLOB_NOSORT` flags.

Reply:
I don't check the return value because it may return only three kind of errors:
1. GLOB_NOSPACE - the system is out of memory. We can abort here, but I don't think this is really necessary. I can add the check if you want.
2. GLOB_ABORTED - the user has no permission to a directory, so it will be skipped. This is useful if the user have no access to certain directories (they will be silently ignored), but have access to the rest of them. I can add the check. Another option would be to specify the mentioned behavior in the docs.
3. GLOB_NOMATCH - because sometimes it is pretty legitimate for a glob not to match anything. See the comment below.

Reply:
> GLOB_NOSPACE - the system is out of memory. We can abort here, but I don't think this is really necessary. I can add the check if you want.

Yeah, this definitely should be checked. Also, systems may invent their own reasons for `glob()` to fail (e.g. now-obsolete `GLOB_NOSYS`).

`GLOB_ABORTED` is impossible if `errfunc == NULL && (flags & GLOB_ERR) == 0`.